src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,114 @@
+        service = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void registerExistingLogin_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin(TEST_LOGIN_ONE);
+        storage.add(existingUser);
+        User newUser = new User();
+        newUser.setLogin(TEST_LOGIN_ONE);
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            service.register(newUser);
+        });
+        assertEquals("User with the same login already exists", exception.getMessage());
+    }
+
+    @Test
+    void registerWithShortLogin_notOk() {
+        User testUser = new User();
+        testUser.setLogin("TEST");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            service.register(testUser);
+        });
+        assertEquals("Login must be at least 6 characters", exception.getMessage());
+    }
+
+    @Test
+    void registerWithShortPassword_notOk() {
+        User testuser = new User();
+        testuser.setLogin(TEST_LOGIN_ONE);
+        testuser.setPassword("abc");
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            service.register(testuser);
+        });
+        assertEquals("Password must be at least 6 characters", exception.getMessage());
+    }
+
+    @Test
+    void registerValidUser_Ok() {
+        User testUser = new User();
+        testUser.setLogin(TEST_LOGIN_ONE);
+        testUser.setPassword(TEST_PASSWORD);
+        testUser.setAge(25);
+        assertDoesNotThrow(() -> service.register(testUser));
+        assertEquals(testUser, storage.get(testUser.getLogin()));
+    }
+
+    @Test
+    void registerLowAge_notOk() {
+        User testUser = new User();
+        testUser.setLogin(TEST_LOGIN_TWO);
+        testUser.setPassword(TEST_PASSWORD);
+        testUser.setAge(15);
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            service.register(testUser);
+        });
+        assertEquals("Age must be at least 18 years old", exception.getMessage());
+    }
+
+    @Test
+    void registerNullLogin_notOk() {
+        User testUser = new User();
+        testUser.setLogin(null);
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            service.register(testUser);
+        });
+        assertEquals("Login must be at least 6 characters", exception.getMessage());
+    }
+
+    @Test
+    void registerNullPassword_notOk() {
+        User testUser = new User();
+        testUser.setLogin(TEST_LOGIN_TWO);
+        testUser.setPassword(null);
+        testUser.setAge(25);
+        InvalidDataException exception = assertThrows(InvalidDataException.class, () -> {
+            service.register(testUser);
+        });
+        assertEquals("Password must be at least 6 characters", exception.getMessage());
+    }
+
+    @Test
+    void registerWithNumbersPasswordAndLogin_Ok() {
```

Comment:
Actually, you don't have any logic about numbers or symbols in login or password, so this test case os redundant. 
It's better to add test case for negative age